src/profile-logic/transforms.js

```diff
@@ -600,3 +599,68 @@ export function focusFunction(
     });
   });
 }
+
+export function restoreAllFunctionsInCallNodePath(
+  thread: Thread,
+  previousImplementationFilter: ImplementationFilter,
+  callNodePath: CallNodePath
+): CallNodePath {
+  const { stackTable, frameTable } = thread;
+  const funcMatchesImplementation = FUNC_MATCHES[previousImplementationFilter];
+  const matchesUpToDepth = [];
+  let tipStackIndex = null;
+  // Try to find the tip most stackIndex in the CallNodePath, but skip anything
+  // that doesn't match the previous implementation filter.
+  for (let stackIndex = 0; stackIndex < stackTable.length; stackIndex++) {
+    const prefix = stackTable.prefix[stackIndex];
+    const frameIndex = stackTable.frame[stackIndex];
+    const funcIndex = frameTable.func[frameIndex];
+    const prefixPathDepth = prefix === null ? -1 : matchesUpToDepth[prefix];
+    if (prefixPathDepth !== null) {
```

Comment:
I'd invert this check and use `continue;`. I had some trouble matching up the `else` below with the right `if`.